ui/controls.js

```diff
@@ -522,6 +522,29 @@ shaka.ui.Controls = class extends shaka.util.FakeEventTarget {
     return this.seekBar_ ? this.seekBar_.getValue() : this.video_.currentTime;
   }
 
+  /**
+   * Checking the efficiency of a video
+   * @return {?number}
+   * @export
+   */
+  getBufferFillPercentage() {
+    const bufferedLength_ = this.video_.buffered.length;
+    const bufferedEnd_ =
+        bufferedLength_ ? this.video_.buffered.end(bufferedLength_ - 1) : 0;
+    const bufferingGoal_ = this.video_.currentTime +
+        Math.min(this.player_.getConfiguration().streaming.bufferingGoal,
```

Comment:
It should work either way, but I would move the Math.min() to the outside and drop the inner subtraction of currentTime.  Something like this looks simpler to me, and therefore easier to read/understand/maintain:

min(currentTime + bufferingGoal, seekRange().end)

Reply:
@joeyparrish I think the name of the variable is a bit confusing, I need to change it. the buffering goal is the value of `currentTime+Min(bufferingGoal, ((seekRange().end -currentTime))  )`  so I could get the minimum between buffering goal and the video time left after the current time because when the video reaches the end and the time left is less than buffering goal the percentage starts to drop but it should be 1. so I took a minimum between the time left and buffering goal.